client/client.cpp

```diff
@@ -21,11 +20,24 @@ Operation genRandOperation() {
 	Operation op;
 	op.__set_op(r < 0.5 ? OpType::PUT : OpType::GET);
   op.__set_key(std::to_string(key));
-	if(op.op == OpType::GET) {
-		char value[VALUE_SIZE];
-		randombytes_buf(value, VALUE_SIZE);
-		op.__set_value(std::string(value));
+
+	std::string value;
+	if (op.op == OpType::GET) {
+		char rand_val[VALUE_SIZE];
+		randombytes_buf(rand_val, VALUE_SIZE);
+		value = std::string(rand_val);
+	} else {
+		int put_val = rand() % VAL_MAX;
+		value = std::to_string(put_val);
 	}
+
+	encryption_engine engine;
+
+	unsigned char* cipher_text = new unsigned char[4096];
+	size_t out_len = (size_t) engine.encryptNonDeterministic(value, cipher_text);
+	std::string updated_val((const char *) cipher_text, out_len);
+
+	op.__set_value(updated_val);
```

Comment:
(question) 
1) Do we only set an updated_val for PUT
2) might have to ask Sujaya but if yes, would an adversarial server not be able to infer the request type based on the size of the packet? I forget what protocol we're using and if this is an issue

Reply:
The updated_val is set for both GET and PUT values. The content is the only difference, where GET is all random bytes and PUT is an actual value. 

The size of the packet should be the same since we pass a `char[4096]` cipher text to host and host receives a `char[4096]` cipher text from enclave as well. Everything stays constant in terms of size